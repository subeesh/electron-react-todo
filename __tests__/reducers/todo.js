import todos from "../../app/reducers/todo";
import * as types from "../../app/constants/todoActionTypes";

test("handles initial state", () => {
  expect(todos(undefined, {})).toEqual([]);
});

test("handles ADD_TODO", () => {
  expect(
    todos([], {
      type: types.ADD_TODO,
      payload: "Learn Redux"
    })
  ).toMatchSnapshot();
});

test("handles DELETE_TODO", () => {
  expect(
    todos(
      [
        {
          text: "Learn React",
          completed: false,
          id: 1
        },
        {
          text: "Learn Redux",
          completed: false,
          id: 0
        }
      ],
      {
        type: types.DELETE_TODO,
        payload: 1
      }
    )
  ).toMatchSnapshot();
});

test("handles EDIT_TODO", () => {
  expect(
    todos(
      [
        {
          text: "Learn Electr",
          completed: false,
          id: 1
        }
      ],
      {
        type: types.EDIT_TODO,
        text: "Learn Electron",
        payload: 1
      }
    )
  ).toMatchSnapshot();
});

test("handles COMPLETE_ALL", () => {
  expect(
    todos(
      [
        {
          text: "Learn ES6",
          completed: true,
          id: 1
        },
        {
          text: "Learn Webpack",
          completed: false,
          id: 0
        }
      ],
      {
        type: types.COMPLETE_ALL
      }
    )
  ).toMatchSnapshot();
});

test("handles CLEAR_COMPLETED", () => {
  expect(
    todos(
      [
        {
          text: "Learn ES6",
          completed: true,
          id: 1
        },
        {
          text: "Learn Electron",
          completed: false,
          id: 0
        }
      ],
      {
        type: types.CLEAR_COMPLETED
      }
    )
  ).toMatchSnapshot();
});
