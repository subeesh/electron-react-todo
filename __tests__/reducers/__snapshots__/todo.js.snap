// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles ADD_TODO 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Learn Redux",
  },
]
`;

exports[`handles CLEAR_COMPLETED 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Learn Electron",
  },
]
`;

exports[`handles COMPLETE_ALL 1`] = `
Array [
  Object {
    "completed": true,
    "id": 1,
    "text": "Learn ES6",
  },
  Object {
    "completed": true,
    "id": 0,
    "text": "Learn Webpack",
  },
]
`;

exports[`handles DELETE_TODO 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Learn Redux",
  },
]
`;

exports[`handles EDIT_TODO 1`] = `
Array [
  Object {
    "completed": false,
    "id": 1,
    "text": "Learn Electr",
  },
]
`;
